Updates to upgrade procedures (#65388)
openshift/openshift-docs: upgrading/upgrade-helm.adoc

toc::[]

[role="_abstract"]
Upgrading {product-title-short} requires a specific upgrade path depending on the release of {product-title-short} that you are running. You must also back up your Central database before updating the Helm chart and peforming the upgrade.

[id="upgrading-from-374_{context}"]
== Upgrade sequence from {product-title-short} release 3.74 and earlier

When upgrading from earlier releases, follow this guidance:

* If the release for Central is earlier than 3.74, you must upgrade to the latest 3.74 patch before upgrading to a 4.x release. See the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74] for information about upgrades from earlier versions to 3.74.

> * When upgrading Helm-based installations from release 3.74, you can upgrade to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to release 4.4.

Author: Created https://issues.redhat.com/browse/ROX-24162 for 4.5+

toc::[]

[role="_abstract"]

> Upgrading {product-title-short} requires a specific upgrade path depending on the release of {product-title-short} that you are running. You must also back up your Central database before updating the Helm chart and peforming the upgrade.

Reviewer: Could make this a bit more active. E.g., "You must follow a specific path to upgrade {product-title-short}..."

[id="upgrading-from-374_{context}"]
== Upgrade sequence from {product-title-short} release 3.74 and earlier

When upgrading from earlier releases, follow this guidance:

* If the release for Central is earlier than 3.74, you must upgrade to the latest 3.74 patch before upgrading to a 4.x release. See the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74] for information about upgrades from earlier versions to 3.74.
* When upgrading Helm-based installations from release 3.74, you can upgrade to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to release 4.4.

If you have installed {product-title-short} by using Helm charts, to upgrade to the latest version of {product-title-short} perform the following steps:

. Back up the Central database.

> . (Optional) Optimize Central database and Persistent Volume Claims (PVC).
> . (Optional) Generate `values-private.yaml` configuration file containing root certificates for the central-services Helm chart.

Reviewer: ```suggestion
. Optional: Optimize central database and Persistent Volume Claims (PVC).
. Optional: Generate `values-private.yaml` configuration file containing root certificates for the central-services Helm chart.
```
or, because it's not a procedure, maybe: 
```suggestion
. Optionally, optimize central database and Persistent Volume Claims (PVC).
. Optionally, generate `values-private.yaml` configuration file containing root certificates for the central-services Helm chart.
```

Author: Central is a component and is always capitalized in the RHACS docs.

toc::[]

[role="_abstract"]

> When upgrading, follow this guidance:

Reviewer 2: I'm nitpicking but this sentence looks very bare bones to me. How about if we make it a sub-section and call it like "upgrade sequence from ACS 3.x"?

Author: This is the abstract, which is a specific field that the Customer Portal requires to provide an overview of what is discussed in the module. I'll add something else for the abstract.

> * If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].

Reviewer 3: Should this be for both Central/SecuredCluster ? 
```suggestion
* If the installed version is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
```

Author: Not sure, I assume so but I think @msugakov had specifically called out Central here, so maybe he can weigh in.

Reviewer 2: Only Central. This is because of rocksdb->postgres database migrations.

@ludydoo if you are aware of the similar need for the Secured Cluster, **this** is a perfect time to mention that!

Reviewer 3: I am not aware of any such thing

Reviewer 2: I suppose we can close this thread as obsolete then.

> * If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].

Reviewer 2: This looks plain wrong to me. Right in the **next bullet point** we tell about upgrading from 3.74 to 4.0-4.4, we don't need to tell the same thing here ... which we actually don't because the provided link is correct.

```suggestion
* If Central is earlier than 3.74, you must upgrade to the latest 3.74 patch before upgrading to a 4.x release. See the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74] regarding upgrades from earlier versions to 3.74.
```

> * For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.

Reviewer 3: ```suggestion
* For Operator- and Helm-based installations, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```

> * For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.

Reviewer 2: 1. We don't need to mention Operator here because these are instructions for Helm.
2. I suggest to make it more explicit that 3.74 -> 4.0..4.4 is ok but 3.74 -> 4.5+ is not ok.
3. I like your point about upgrading to the latest but I suggest to reword it slightly so that the proposed action is clear.

```suggestion
* When upgrading Helm-based Central deployments from release 3.74, choose any latest patch of {product-title-short} version 4.0 through 4.4 but not 4.5 or later. After that, upgrade Central to the latest version.
```

Also, a link doc from 4.4's documentation would be useful.

Author: > 2\. I suggest to make it more explicit that 3.74 -> 4.0..4.4 is ok but 3.74 -> 4.5+ is not ok.

This PR isn't for 4.5, this is for 4.4 and earlier. I'll make a separate PR for 4.5.

Author: Rewrote because the intent was to tell them to upgrade to 4.4.x (the latest version, as of this publication - 4.4 version).

Author: And I can add a link to the 4.4 docs from the 4.5 docs if you think that's necessary. (per your comment above)

> * For upgrading within version 4.x, see the latest version of the documentation.

Reviewer 2: ```suggestion
```
Because this page _is_ the latest version of the documentation.
